src/chrome/chromeDebugAdapter.ts

```diff
@@ -646,75 +655,88 @@ export abstract class ChromeDebugAdapter implements IDebugAdapter {
         this._lineColTransformer.columnBreakpointsEnabled = this._columnBreakpointsEnabled;
     }
 
-    protected async onScriptParsed(script: Crdp.Debugger.ScriptParsedEvent): Promise<void> {
-        this.doAfterProcessingSourceEvents(async () => { // This will block future 'removed' source events, until this processing has been completed
-            if (typeof this._columnBreakpointsEnabled === 'undefined') {
-                await this.detectColumnBreakpointSupport(script.scriptId).then(async () => {
-                    await this.sendInitializedEvent();
-                });
-            }
-
-            if (this._earlyScripts) {
-                this._earlyScripts.push(script);
-            } else {
-                await this.sendLoadedSourceEvent(script);
-            }
-        });
-
-        if (script.url) {
-            script.url = utils.fixDriveLetter(script.url);
+    public breakpointsResolvedDefer(scriptId: string): PromiseDefer<void> {
+        const existingValue =  this._scriptIdToBreakpointsAreResolvedDefer.get(scriptId);
+        if (existingValue) {
+            return existingValue;
         } else {
-            script.url = ChromeDebugAdapter.EVAL_NAME_PREFIX + script.scriptId;
+            const newValue = promiseDefer<void>();
+            this._scriptIdToBreakpointsAreResolvedDefer.set(scriptId, newValue);
+            return newValue;
         }
+    }
+
+    protected async onScriptParsed(script: Crdp.Debugger.ScriptParsedEvent): Promise<void> {
+        const breakpointsAreResolvedDefer = this.breakpointsResolvedDefer(script.scriptId);
+        try {
+            this.doAfterProcessingSourceEvents(async () => { // This will block future 'removed' source events, until this processing has been completed
+                if (typeof this._columnBreakpointsEnabled === 'undefined') {
+                    await this.detectColumnBreakpointSupport(script.scriptId).then(async () => {
```

Comment:
Don't mix async/await and promises when possible

Reply:
if (typeof this._columnBreakpointsEnabled === 'undefined') {
                    await this.detectColumnBreakpointSupport(script.scriptId);
                    await this.sendInitializedEvent();
                }